modules/tempo/main.tf

```diff
@@ -0,0 +1,90 @@
+resource "helm_release" "tempo" {
+  name             = "tempo"
+  repository       = "https://grafana.github.io/helm-charts"
+  chart            = "tempo"
+  version          = "1.18.3"
+  namespace        = var.namespace
+  create_namespace = true
+
+  values = [
+    templatefile("${path.module}/values/tempo-values.yaml.tpl", {
+      tempo_image_tag = var.configs.tempo_image_tag
+      storage_backend = var.configs.storage_backend
+      bucket_name     = module.my_bucket.s3_bucket_id
+      region          = var.region
+
+      persistence_enabled = var.configs.persistence.enabled
+      persistence_size    = var.configs.persistence.size
+      persistence_class   = var.configs.persistence.storage_class
+
+      enable_metrics_generator = var.configs.enable_metrics_generator
+      enable_service_monitor   = var.configs.enable_service_monitor
+
+      enable_ingress      = var.configs.ingress.enabled
+      ingress_annotations = var.configs.ingress.annotations
+      ingress_hosts       = var.configs.ingress.hosts
+      ingress_path        = var.configs.ingress.path
+      ingress_path_type   = var.configs.ingress.path_type
+
+      service_account_name        = var.configs.service_account.name
+      service_account_annotations = local.service_account_annotations
+    })
+  ]
+  depends_on = [module.my_bucket]
+}
+
+module "my_bucket" {
+  source  = "dasmeta/s3/aws"
+  version = "1.2.1"
+
+  name = var.configs.bucket_name
+
+
+  acl = "public"
```

Comment:
1. why we have this bucket public?
this bucket tempo uses for its internal storage, can we have it private, the default one is private so maybe we can just remove this acl line?
2. also there is available new version 1.3.1 for dasmeta/s3/aws module, will be great if we use latest versions of modules, if there is no special need to use older version,
3. can we have better name for this module, `my_bucket` sounds more like a test tmp name for tf modules